hs_core/views/utils.py

```diff
@@ -57,18 +58,22 @@ def authorize(request, res_id, edit=False, view=False, full=False, superuser=Fal
     except ObjectDoesNotExist:
         raise NotFound(detail="No resource was found for resource id:%s" % res_id)
 
-    if user.is_authenticated():
+    if discoverable and (edit is False and view is False and full is False and superuser is False):
+        authorized = res.raccess.discoverable
+    elif user.is_authenticated():
         has_edit = user.uaccess.can_change_resource(res)
         has_view = user.uaccess.can_view_resource(res)
         has_full = user.uaccess.owns_resource(res)
 
-        authorized = (edit and has_edit) or \
+        authorized = (discoverable and res.raccess.discoverable) or \
+                     (edit and has_edit) or \
                      (view and has_view) or \
                      (full and has_full) or \
                      (superuser and (user.uaccess.admin or user.is_superuser))
     else:
         has_view = res.raccess.public
-        authorized = view and has_view
+        authorized = (discoverable and res.raccess.discoverable) or \
+                     (view and has_view)
```

Comment:
@selimnairb Yeah, agreed we need to distinguish these two cases, but since this authorize() function only returns true or false, you will have to do additional check anyway after calling this authorize() function, so I think perhaps this authorize() function just does a bit more relaxed check for discoverable case, then an addition check can be added when specifically you want to determine whether to expose content files or not, or perhaps there is a better approach to make this function return a bit of more info so you don't have to do extra check afterwards?